InfiniteScrollView.js

```diff
@@ -70,10 +89,22 @@ class InfiniteScrollView {
     }
 
     if (this._distanceFromEnd(event) < this.props.distanceToLoadMore) {
-      this.props.onLoadMore();
+      this._onLoadMore();
     }
   }
 
+  _onLoadMore() {
+    this.props.onLoadMoreAsync().catch((error) => {
```

Comment:
Now that onLoadMore is a promise, I think we can get rid of `isLoading` since that's implied by whether the promise has settled. Can you think of a case where we'd want to set `isLoading` to false even though the promise is still in flight?

Where this gets most complicated is if `onLoadMoreAsync` is invoked twice without waiting for the promise from the first invocation to complete. There's actually a potential bug here since with the current API you could set `isLoading=false` even if there's an in-flight request, then InfiniteScrollView would initiate another request, and now lets say the second request comes back successful but the first fails. Seems simpler if we avoid all that by limiting InfiniteScrollView to one request at a time if we believe that is a sufficiently flexible API.

Style nit: leave off the parens around `error`.